pinot-common/src/main/java/org/apache/pinot/common/assignment/InstancePartitions.java

```diff
@@ -119,6 +120,7 @@ public void setInstances(int partitionId, int replicaGroupId, List<String> insta
   }
 
   public static InstancePartitions fromZNRecord(ZNRecord znRecord) {
+    Preconditions.checkArgument(znRecord != null, "ZNRecord cannot be null");
```

Comment:
If we are passing null here from some call site, it might be better to also update the call-site so it can log the relevant details and perhaps avoid the call itself?

Seeing NPE or IllegalArgumentException doesn't really change much from a debuggability perspective.